connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala

```diff
@@ -194,14 +195,16 @@ case class SessionHolder(userId: String, sessionId: String, session: SparkSessio
    */
   private[service] def interruptTag(tag: String): Seq[String] = {
     val interruptedIds = new mutable.ArrayBuffer[String]()
+    val queries = SparkConnectService.streamingSessionManager.getTaggedQuery(tag, session)
+    queries.foreach(_.query.stop())
```

Comment:
This one, I am not sure if we should just simply stop with blocking, which is simple and straightforward but not sure if we should do something else, @WweiL

Reply:
By default `stop` is a blocking operation because we set the timeout to be 0 here:
https://github.com/apache/spark/blob/94763438943e21ee8156a4f1a3facddbf8d45797/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala#L474-L489

Reply:
I think the semantics of the interrupt is purely just interrupt but no wait:
https://github.com/apache/spark/blob/master/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala#L180-L188

My understanding is that if there is some long running batch query (like in classic spark, do a spark.range(1e10) and it blocks the main thread, then we stop that execution, and don't care about the remaining additional remaining work).

Then does it make sense to do something similar to the streaming query? Imagine the streaming query thread to be the main thread, and it is likely currently executing some microbatch that takes long, and we just send `interrupt` signal and ignores what is left over (cleanup, etc).

Reply:
But seems like we cancel jobs in that case:

```
  override def stop(): Unit = {
    // Set the state to TERMINATED so that the batching thread knows that it was interrupted
    // intentionally
    state.set(TERMINATED)
    if (queryExecutionThread.isAlive) {
      sparkSession.sparkContext.cancelJobGroup(runId.toString)
      interruptAndAwaitExecutionThreadTermination()
      // microBatchThread may spawn new jobs, so we need to cancel again to prevent a leak
      sparkSession.sparkContext.cancelJobGroup(runId.toString)
    }
    logInfo(log"Query ${MDC(LogKeys.PRETTY_ID_STRING, prettyIdString)} was stopped")
  }
```

so I think we're all good

Reply:
To avoid blocking call, we can just do `new Thread(() => q.stop())`